### ISSUE TITLE
provider not able to update always_pxe

### ISSUE THREAD

**Issue by ae-v** [terraform-providers/terraform-provider-packet#56]
### Terraform Version
Terraform v0.11.6
+ provider.packet v1.2.0

### Affected Resource(s)
Please list the resources as a list, for example:
- packet_device

### Terraform Configuration Files
```hcl
resource "packet_device" "pxe-node" {
  hostname         = "foo"
  plan             = "baremetal_2"
  facility         = "ewr1"
  operating_system = "custom_ipxe"
  always_pxe       = "true"
  billing_cycle    = "hourly"
  project_id       = "foo-id"
  ipxe_script_url  = "http://matchbox.foo.wtf:8080/boot.ipxe"
}
```

### Expected Behavior
An update of the attribute is done on a packet_device. Throw an error when applying if the update fails.

### Actual Behavior
Terraform neither updates the device nor the state.

**Comment by bzub** [comment]
Thanks for reporting this @ae-v. I've opened an issue in the packngo repo where the bug appears to originate: https://github.com/packethost/packngo/issues/66

**Comment by t0mk** [comment]
hi, thanks for reporting @ae-v and thanks for fixing the culprit @bzub.

I merged the fix to packngo. Later today I try to bump packngo in the provider and verify this issue. There's been a couple of changes in packngo, so it might take a longer time to integrate.

**Comment by t0mk** [comment]
@ae-v Can you please verify that this is fixed?

**Comment by ae-v** [comment]
Yes, it doesn't work with terraform-provider-packet_v1.2.1.

I tried to set a device from always_pxe: "true" => "false" but it's neither written to state nor updates the device at Packet.

**Comment by t0mk** [comment]
ok, that's strange, as I wrote a test case exactly for this: https://github.com/terraform-providers/terraform-provider-packet/blob/master/packet/resource_packet_device_test.go#L120

I will test with HCL from your first comment, and then change the always_pxe to false.

**Comment by t0mk** [comment]
I am going to sleep now, so I mark this here - the culprit seems to be:
```
	if !reflect.DeepEqual(ur, packngo.DeviceUpdateRequest{}) {
		if _, _, err := client.Devices.Update(d.Id(), &ur); err != nil {
			return friendlyError(err)
		}

	}
```

Because `false` is a default for `always_pxe`, the DeepEqual with the default UpdateRequest is true (if  the only attr changed is `always_pxe`), and thus HTTP call is skipped. 

This is going to happen for any other single field update to the fields default.

Temporary workaround: change something else together with `always_pxe`. If you change it to false, maybe you can afford to change ipxe_script_url to "https://dummy.url". Then the device resource will update.

I will try to fix the update tomorrow (17.4). Thanks for patience.

**Comment by ae-v** [comment]
Now it works in both directions. Great! Thanks a lot for your effort.
